src/main/java/ome/services/fulltext/FullTextBridge.java

```diff
@@ -100,15 +101,41 @@ public FullTextBridge(OriginalFilesService files,
      *            set of {@link FieldBridge bridge classes} which will be
      *            instantiated via a no-arg constructor.
      * @see <a
-     *      href="https://trac.openmicroscopy.org.uk/ome/SearchBridges">Bridges</a
+     *      href="https://omero.readthedocs.io/en/stable/developers/Modules/Search/Bridges.html">Bridges</a
      *      href>
      */
     @SuppressWarnings("unchecked")
     public FullTextBridge(OriginalFilesService files,
             Map<String, FileParser> parsers, Class<FieldBridge>[] bridgeClasses) {
+        this(files, parsers, bridgeClasses, 1);
+    }
+
+    /**
+     * Main constructor.
+     *
+     * @param files
+     *            {@link OriginalFilesService} for getting access to binary files.
+     * @param parsers
+     *            List of {@link FileParser} instances which are currently
+     *            configured.
+     * @param bridgeClasses
+     *            set of {@link FieldBridge bridge classes} which will be
+     *            instantiated via a no-arg constructor.
+     * @param maxFilesetSize
+     *            maximum size of the fileset to be considered for indexing
+     * @see <a
+     *      href="https://omero.readthedocs.io/en/stable/developers/Modules/Search/Bridges.html">Bridges</a
+     *      href>
+     */
+    @SuppressWarnings("unchecked")
+    public FullTextBridge(OriginalFilesService files,
+            Map<String, FileParser> parsers, Class<FieldBridge>[] bridgeClasses,
+            int maxFilesetSize) {
         this.files = files;
         this.parsers = parsers;
         this.classes = bridgeClasses == null ? new Class[] {} : bridgeClasses;
+        this.maxFilesetSize = maxFilesetSize;
```

Comment:
default in the constructor above is set to ``1``.
This could allow to pass any value including negative one. Maybe add a check

Reply:
Do you and/or anyone else who was involved in the initial review of the feature foresee a situation where someone would like to preserve the current behavior i.e. index all files in a fileset irrespectively of its size?

If so, `-1` (or any negative value?) could be used as a mechanism to skip the fileset size check, with all associated warnings re performance implications. Otherwise, I can make this default to 1 if it's negative.

Reply:
I wasn't involved in the initial review or discussion around the feature back in late 2019. However, purely from a technical perspective I can see justification for leaving this open ended. If we add range checks here in the constructor they will apply to all subclasses and that seems like a bad design decision. There are other ways of doing this with Spring such as using properties rather than constructor arguments but I don't know if that's needed here.